[Bug] Using `once` record_mode gives error

# Problem

Using `once` gives an error:

```sh
vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ([PATH_OF_CASETTE_FILE]) in your current record mode ('once').
E               No match for the request (<Request (GET) https://postman-echo.com/get?foo1=bar1&foo2=bar2>) was found.
E               No similar requests, that have not been played, found.
```

I ran this the first time so PATH_OF_CASETTE_FILE shouldn't exist although the error seems to say it exists. I checked the PATH_OF_CASETTE_FILE that the error specifies *before* I run my test the 2nd time and indeed no such file exists.

# To Reproduce

```
import pytest
import requests

@pytest.mark.vcr(record_mode="once")
def test_vcr():
    response = requests.get("https://postman-echo.com/get?foo1=bar1&foo2=bar2")
    assert response.status_code == 200
```

# Stacktrace
<details>
  <summary>Click to expand!</summary>

```sh
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/requests/api.py:76: in get
    return request('get', url, params=params, **kwargs)
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/requests/api.py:61: in request
    return session.request(method=method, url=url, **kwargs)
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/requests/sessions.py:530: in request
    resp = self.send(prep, **send_kwargs)
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/requests/sessions.py:643: in send
    r = adapter.send(request, **kwargs)
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/requests/adapters.py:439: in send
    resp = conn.urlopen(
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:670: in urlopen
    httplib_response = self._make_request(
../../Library/Caches/pypoetry/virtualenvs/featuredproductsswitch-q2O7GQgj-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:417: in _make_request
    httplib_response = conn.getresponse(buffering=True)
```
</details>

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -6,6 +6,11 @@ Changelog
 `Unreleased`_
 -------------
 
+Fixed
+~~~~~
+
+- Honor ``record_mode`` passed via ``pytest.mark.vcr`` mark or in ``vcr_config`` fixture. `#47`_
+
 `0.8.0`_ - 2020-06-06
 ---------------------
 
@@ -134,6 +139,7 @@ Added
 .. _0.3.0: https://github.com/kiwicom/pytest-recording/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/kiwicom/pytest-recording/compare/v0.1.0...v0.2.0
 
+.. _#47: https://github.com/kiwicom/pytest-recording/issues/47
 .. _#45: https://github.com/kiwicom/pytest-recording/issues/45
 .. _#37: https://github.com/kiwicom/pytest-recording/issues/37
 .. _#34: https://github.com/kiwicom/pytest-recording/issues/34

--- a/src/pytest_recording/_vcr.py
+++ b/src/pytest_recording/_vcr.py
@@ -35,15 +35,21 @@ class CombinedPersister(FilesystemPersister):
         #    but the functional approach is faster
         # 2. It could be done more efficient, but the `deserialize` implementation should be adjusted as well
         #    But it is a private API, which could be changed.
-        return starmap(unpack, zip(*all_content))
+        requests, responses = starmap(unpack, zip(*all_content))
+        requests, responses = list(requests), list(responses)
+        if not requests or not responses:
+            raise ValueError("No cassettes found.")
+        return requests, responses
 
 
 # pylint: disable=too-many-arguments
 def use_cassette(default_cassette, vcr_cassette_dir, record_mode, markers, config, pytestconfig):
     """Create a VCR instance and return an appropriate context manager for the given cassette configuration."""
     merged_config = merge_kwargs(config, markers)
+    if "record_mode" in merged_config:
+        record_mode = merged_config["record_mode"]
     path_transformer = get_path_transformer(merged_config)
-    if record_mode == "rewrite" or merged_config.get("record_mode") == "rewrite":
+    if record_mode == "rewrite":
         path = path_transformer(os.path.join(vcr_cassette_dir, default_cassette))
         try:
             os.remove(path)

--- a/tests/test_recording.py
+++ b/tests/test_recording.py
@@ -43,6 +43,46 @@ def test_cassette_recording(testdir):
     assert not cassette_path.exists()
 
 
+def test_record_mode_in_mark(testdir):
+    # See GH-47
+    testdir.makepyfile(
+        """
+        import pytest
+        import requests
+
+        @pytest.mark.vcr(record_mode="once")
+        def test_record_mode(httpbin):
+            assert requests.get(httpbin.url + "/get").status_code == 200
+    """
+    )
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=1)
+    cassette_path = testdir.tmpdir.join("cassettes/test_record_mode_in_mark/test_record_mode.yaml")
+    assert cassette_path.size()
+
+
+def test_record_mode_in_config(testdir):
+    # See GH-47
+    testdir.makepyfile(
+        """
+        import pytest
+        import requests
+
+        @pytest.fixture(scope="module")
+        def vcr_config():
+            return {"record_mode": "once"}
+
+        @pytest.mark.vcr
+        def test_record_mode(httpbin):
+            assert requests.get(httpbin.url + "/get").status_code == 200
+    """
+    )
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=1)
+    cassette_path = testdir.tmpdir.join("cassettes/test_record_mode_in_config/test_record_mode.yaml")
+    assert cassette_path.size()
+
+
 def test_cassette_recording_rewrite(testdir):
     testdir.makepyfile(
         """

